Core/Core/Kits/Units.cs

```diff
@@ -234,109 +236,67 @@ public static double GetConversionFactor(string from, string to)
     return 1;
   }
 
-  public static string GetUnitsFromString(string unit)
+  /// <summary>
+  ///
+  /// </summary>
+  /// <param name="unit"></param>
+  /// <returns></returns>
+  /// <exception cref="ArgumentOutOfRangeException">Unit string is not a supported unit (see <see cref="IsUnitSupported"/>)</exception>
+  [Pure]
+  public static string? GetUnitsFromString(string unit)
   {
     if (unit == null)
     {
       return null;
     }
 
-    switch (unit.ToLower())
+    return unit.ToLower() switch
     {
-      case "mm":
-      case "mil":
-      case "millimeter":
-      case "millimeters":
-      case "millimetres":
-        return Millimeters;
-      case "cm":
-      case "centimetre":
-      case "centimeter":
-      case "centimetres":
-      case "centimeters":
-        return Centimeters;
-      case "m":
-      case "meter":
-      case "metre":
-      case "meters":
-      case "metres":
-        return Meters;
-      case "inches":
-      case "inch":
-      case "in":
-        return Inches;
-      case "feet":
-      case "foot":
-      case "ft":
-        return Feet;
-      case "ussurveyfeet":
-        return USFeet;
-      case "yard":
-      case "yards":
-      case "yd":
-        return Yards;
-      case "miles":
-      case "mile":
-      case "mi":
-        return Miles;
-      case "kilometers":
-      case "kilometer":
-      case "km":
-        return Kilometers;
-      case "none":
-        return None;
-    }
-
-    throw new SpeckleException($"Cannot understand what unit {unit} is.");
+      "mm" or "mil" or "millimeter" or "millimeters" or "millimetres" => Millimeters,
+      "cm" or "centimetre" or "centimeter" or "centimetres" or "centimeters" => Centimeters,
+      "m" or "meter" or "metre" or "meters" or "metres" => Meters,
+      "inches" or "inch" or "in" => Inches,
+      "feet" or "foot" or "ft" => Feet,
+      "ussurveyfeet" => USFeet, //BUG: why don't we match on "us_ft"?
+      "yard" or "yards" or "yd" => Yards,
+      "miles" or "mile" or "mi" => Miles,
+      "kilometers" or "kilometer" or "km" => Kilometers,
+      "none" => None,
+      _ => throw new ArgumentOutOfRangeException(nameof(unit), $"Unrecognised unit string {unit}"),
```

Comment:
can I love and hate this at the same time? 😅

That BUG comment is legit, good catch @JR-Morgan 👀 @clairekuang maybe you know the reason?

Reply:
You'd prefer the switch statement over the switch expression? (I disagree, but not hard enough to object)
Or simply you dislike the fact we matching several raw string values to another raw string value... (this I agree is smelly)

Reply:
As for us feet, I have a feeling the us_ft are not being used by us, so we may want to flag as [Obsolete].

Reply:
> Or simply you dislike the fact we matching several raw string values to another raw string value

this ☝🏼 hehe switch expressions are lovable things ♥️